factory/typescript/notify.interface.ts

```diff
@@ -0,0 +1,5 @@
+import {User} from "./user";
+
```

Comment:
UML diyagramında denk geldiği yerleri göstermek için genelde yorumlar bırakıyorum, buraya da bırakabiliriz tıpkı diğer örneklerde olduğu gibi.

// UML diyagramındaki IProduct arayüzümüze denk gelmektedir.
// İlişkili sınıflarımız bu arayüzü uygulayacaktır.